File: ems_hr_web/src/main/webapp/js/service/staffRapService.js
app.service('staffRapService',function ($http) {

   //分页查询奖惩记录
    this.search=function (page,rows,staffRap) {
        return $http.post('../staffRap/search.do?page='+page+"&rows="+rows,staffRap);
    }

    //查询单条记录
    this.findOne=function (id) {
        return $http.get('../staffRap/findOne.do?id='+id);
    }

    //增加
    this.add=function(entity){
        return  $http.post('../staffRap/add.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../staffRap/update.do',entity );
    }

    //删除
    this.dele=function (ids) {
        return $http.get('../staffRap/delete.do?ids='+ids);
    }
});

File: ems_interface/src/main/java/com/ems/service/StaffRapService.java
package com.ems.service;

import com.ems.pojo.StaffRap;
import entity.PageResult;

import java.util.List;

/**
 * @InterfaceName: StaffRapService
 * @Description: TODO
 * @Autor:13528
 * @Date: 2020/3/25 10:57
 * @Version 1.0
 **/
public interface StaffRapService {

    public PageResult search(StaffRap staffRap, int page, int rows);

    public StaffRap findOne(Integer id);

    public void add(StaffRap staffRap);

    public void update(StaffRap staffRap);

    public void dele(Integer[] ids);

    public List<StaffRap> findById(Integer id);
}
